gum/arch-arm64/gumarm64writer.c

```diff
@@ -1992,15 +1993,37 @@ gum_arm64_writer_commit_literals (GumArm64Writer * self)
     if (r->width != GUM_LITERAL_64BIT)
       continue;
 
+    /*
+     * Whilst instructions in aarch64 are always in little endian (even on
+     * big-endian systems), the data is in native endian. Thus since we wish to
+     * support writing code for big-endian systems on little-endian targets and
+     * vice versa, we need to check the writer configuration.
+     */
     for (slot = first_slot; slot != last_slot; slot++)
     {
-      if (GINT64_FROM_LE (*slot) == r->val)
-        break;
+      if (self->data_endian == GUM_ENDIAN_LITTLE)
```

Comment:
Is the optimizer able to avoid repeatedly reading and checking this field's value, or should we introduce a local variable used throughout the function?

Reply:
I would be very surprised if the optimizer didn't take care of it anyway, but the actual code generation part isn't hugely performance critical (to the point where a few cycles matter), its the performance of the generated code which is most important (for applications such as fuzzing).